lib/flight_scheduler/job_registry.rb

```diff
@@ -116,6 +116,29 @@ def running_tasks_for(job)
     end
   end
 
+  def load
+    data = persistence.load
+    return if data.nil?
+    task_hashes, job_hashes = data.partition do |h|
+      h['job_type'] == 'ARRAY_TASK'
+    end
+    (job_hashes + task_hashes).each do |hash|
+      job = Job.from_serialized_hash(hash)
+      if job.valid?
+        add(job)
+      else
+        Async.logger.warn("Invalid job loaded: #{job.errors.inspect}")
+      end
+    end
+  rescue
+    Async.logger.warn("Error loading job registry: #{$!.message}")
+    raise
+  end
+
+  def save
+    persistence.save(jobs.map(&:serializable_hash))
```

Comment:
Probably best to wrap this in a `read_lock`

Reply:
The `jobs` method is already wrapped in a `read_lock`.  If I add one here too, an exception will be raised.  I could operate on `@jobs` directly, but then I'd be duplicating `jobs` in this method.